packages/@aws-cdk/aws-appmesh/lib/virtual-gateway.ts

```diff
@@ -103,6 +119,56 @@ abstract class VirtualGatewayBase extends cdk.Resource implements IVirtualGatewa
       virtualGateway: this,
     });
   }
+
+  /**
+   * Grants the given entity all read permissions for this VirtualGateway.
+   */
+  public grantRead(identity: iam.IGrantable): iam.Grant {
+    return this.grant(identity,
+      'appmesh:DescribeVirtualGateway',
+      'appmesh:ListVirtualGateway',
+      'appmesh:DescribeGatewayRoute',
+      'appmesh:ListGatewayRoute',
+    );
+  }
+
+  /**
+   * Grants the given entity all write permissions for this VirtualGateway.
+   */
+  public grantWrite(identity: iam.IGrantable): iam.Grant {
+    return this.grant(identity,
+      'appmesh:CreateVirtualGateway',
+      'appmesh:UpdateVirtualGateway',
+      'appmesh:DeleteVirtualGateway',
+      'appmesh:CreateGatewayRoute',
+      'appmesh:UpdateGatewayRoute',
+      'appmesh:DeleteGatewayRoute',
+      'appmesh:TagResource',
+      'appmesh:UntagResource',
+    );
+  }
+
+  /**
+   * Grants the given entity `appmesh:StreamAggregatedResources`.
+   */
+  public grantStreamAggregatedResources(identity: iam.IGrantable): iam.Grant {
+    return iam.Grant.addToPrincipal({
+      grantee: identity,
+      actions: ['appmesh:StreamAggregatedResources'],
+      resourceArns: [this.virtualGatewayArn],
+    });
+  }
+
+  /**
+   * Grant the specified actions for this VirtualGateway.
+   */
+  private grant(grantee: iam.IGrantable, ...actions: string[]) {
+    return iam.Grant.addToPrincipal({
+      grantee,
+      actions,
+      resourceArns: [this.virtualGatewayArn, this.virtualGatewayArn + '/gatewayRoute/*'],
```

Comment:
Same question here (can we divide the permissions between the 2 resources that are needed for them?).